test/cql-pytest/test_materialized_view.py

```diff
@@ -369,3 +369,73 @@ def test_oversized_base_regular_view_key_build(cql, test_keyspace, cassandra_bug
                         break
                 time.sleep(0.1)
             assert results == {(str(i), i) for i in range(30)}
+
+# Reproduces #11668
+# When the view builder resumes building a partition, it reuses the reader
+# used from the previous step but re-creates the compactor. This means that any
+# range tombstone changes active at the time of suspending the step, have to be
+# explicitly re-opened on when resuming. Without that, already deleted base rows
+# can be resurrected as demonstrated by this test.
+def test_view_builder_suspend_with_active_range_tombstone(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "pk int, ck int, v int, PRIMARY KEY(pk, ck)", "WITH compaction = {'class': 'NullCompactionStrategy'}") as table:
+        stmt = cql.prepare(f'INSERT INTO {table} (pk, ck, v) VALUES (?, ?, ?)')
+
+        # sstable 1 - even rows
+        for ck in range(0, 512, 2):
+            cql.execute(stmt, (0, ck, ck))
+        nodetool.flush(cql, table)
+
+        # sstable 2 - odd rows and a range tombstone covering even rows
+        # we need two sstables so memtable doesn't compact away the shadowed rows
+        cql.execute(f"DELETE FROM {table} WHERE pk = 0 AND ck >= 0 AND ck < 512")
+        for ck in range(1, 512, 2):
+            cql.execute(stmt, (0, ck, ck))
+        nodetool.flush(cql, table)
+
+        # we should not see any even rows here - they are covered by the range tombstone
+        res = [r.ck for r in cql.execute(f"SELECT ck FROM {table} WHERE pk = 0")]
+        assert res == list(range(1, 512, 2))
+
+        with new_materialized_view(cql, table, select='*', pk='v,pk,ck', where='v is not null and pk is not null and ck is not null') as mv:
+            start_time = time.time()
+            while time.time() < start_time + 30:
+                res = sorted([r.v for r in cql.execute(f"SELECT * FROM {mv}")])
+                if len(res) >= 512/2:
+                    break
+                time.sleep(0.1)
+            # again, we should not see any even rows in the materialized-view,
+            # they are covered with a range tombstone in the base-table
+            assert res == list(range(1, 512, 2))
+
+# A variant of the above using a partition-tombstone, which is also lost similar
+# to range tombstones.
+def test_view_builder_suspend_with_partition_tombstone(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "pk int, ck int, v int, PRIMARY KEY(pk, ck)", "WITH compaction = {'class': 'NullCompactionStrategy'}") as table:
```

Comment:
@denesb both these test fails on Cassandra (`test/cql-pytest/run-cassandra test_materialized_views.py::test_view_builder_suspend_with_partition_tombstone`) because `NullCompactionStrategy` doesn't seem to exist in Cassandra. Is the null compaction strategy necessary for this test?
If it is we have two options: The easier option is just to mark this test scylla_only, as it can't run on Cassandra. The harder way (maybe isn't worth your time) is to write a function that disables compaction on a table on both Scylla and Cassandra: In Scylla, alter the table to use NullCompactionStrategy. In Cassandra, use "nodetool disableautocompaction".

Reply:
What the test requires is that compaction is disabled and using `NullCompactionStrategy` seemed like the easiest option to achieve that in a targeted manner. The reason I need this is because I need to have 2 sstables whose content I control, without having to worry about compaction intervening and compacting them, nullifying my test.

Reply:
So I think it will be easiest to just add scylla_only. This test reproduces a bug, not a feature, so it is "obvious" it tests the right thing even without double-checking on Cassandra.